src/inputs/components/input-types/duration-input.js

```diff
@@ -0,0 +1,124 @@
+import React, { Component } from 'react';
+import Select from 'react-select';
+import moment from 'moment';
+
+const UNITS = [
+    'seconds',
+    'minutes',
+    'hours',
+    'days',
+    'weeks',
+    'months',
+    'years'
+];
+
+const UNITS_FOR_SELECT = UNITS.map((unit) => {
+    return { value: unit, label: unit };
+});
+
+// Do a best effort converting a moment.duration
+// into a numeric value and unit value.
+// Start with the largest unit and down
+// until the duration is a multiple of the unit.
+export function getBestNumValueAndUnit(duration) {
+    let bestUnit = 'minutes';
+    if (duration === null) {
+        return {
+            numValue: null,
+            unit: bestUnit
+        };
+    }
+
+    for (let i = UNITS.length - 1; i >= 0; i--) {
+        if (duration.as(UNITS[i]) % 1 === 0) {
+            bestUnit = UNITS[i];
+            break;
+        }
+    }
+
+    return {
+        numValue : duration.as(bestUnit),
+        unit : bestUnit
+    };
+}
```

Comment:
I would suggest not export this function and seeing about unit testing the react component as a whole. It's complicated enough to warrant it. Looks like its not too bad with jest: https://facebook.github.io/jest/docs/tutorial-react.html